src/frontend/src/lib/components/guard/AddressGuard.svelte

```diff
@@ -1,29 +1,49 @@
 <script lang="ts">
 	import { addressStore } from '$lib/stores/address.store';
-	import { isNullish } from '@dfinity/utils';
-	import { certifyEthAddress } from '$lib/services/address.services';
+	import { isNullish, nonNullish } from '@dfinity/utils';
+	import { certifyBtcAddressMainnet, certifyEthAddress } from '$lib/services/address.services';
 	import { warnSignOut } from '$lib/services/auth.services';
-	import { ETHEREUM_TOKEN_ID } from '$env/tokens.env';
+	import {
+		btcAddressMainnet,
+		btcAddressMainnetData,
+		btcAddressMainnetNotCertified,
+		ethAddress,
+		ethAddressData,
+		ethAddressNotCertified
+	} from '$lib/derived/address.derived';
+	import { NETWORK_BITCOIN_ENABLED } from '$env/networks.btc.env.js';
+	import type { ResultSuccess } from '$lib/types/utils';
 
 	const validateAddress = async () => {
-		const ethAddressData = $addressStore?.[ETHEREUM_TOKEN_ID];
-
-		if (isNullish(ethAddressData)) {
+		if (isNullish($btcAddressMainnetData) && isNullish($ethAddressData)) {
 			// No address is loaded, we don't have to verify it
 			return;
 		}
 
-		const { certified, data: ethAddress } = ethAddressData;
-
-		if (certified === true) {
-			// The address is certified, all good
+		if (!$btcAddressMainnetNotCertified && !$ethAddressNotCertified) {
+			// The addresses are certified, all good
 			return;
 		}
 
-		const { success, err } = await certifyEthAddress(ethAddress);
+		const results = await Promise.all([
+			NETWORK_BITCOIN_ENABLED && nonNullish($btcAddressMainnet) && $btcAddressMainnetNotCertified
+				? certifyBtcAddressMainnet($btcAddressMainnet)
+				: Promise.resolve<ResultSuccess<never>>({ success: true }),
+			nonNullish($ethAddress) && $ethAddressNotCertified
+				? certifyEthAddress($ethAddress)
+				: Promise.resolve<ResultSuccess<never>>({ success: true })
+		]);
+
+		const { success, err } = results.reduce(
```

Comment:
we use it after in case of non success:

```typescript
if (success) {
	// The addresses are valid
	return;
}

await warnSignOut(err ?? 'Error while certifying your address');
```

Reply:
Ok. In this or another PR, can we have a function for this? I think there are two similar reduce over the results in recent PRs.

Reply:
Will do!

Reply:
doing in PR https://github.com/dfinity/oisy-wallet/pull/2120

Reply:
adapted